lib/rucio/core/account_limit.py

```diff
@@ -130,70 +138,33 @@ def get_global_account_limits(account: Optional["InternalAccount"] = None, *, se
 
 
 @read_session
-def get_global_account_limit(account: "InternalAccount", rse_expression: str, *, session: "Session") -> Union[int, float, None]:
-    """
-    Returns the global account limit for the account on the rse expression.
-
-    :param account:         Account to check the limit for.
-    :param rse_expression:  RSE expression to check the limit for.
-    :param session:         Database session in use.
-    :return:                Limit in Bytes.
-    """
-    try:
-        stmt = select(
-            models.AccountGlobalLimit
-        ).where(
-            and_(models.AccountGlobalLimit.account == account,
-                 models.AccountGlobalLimit.rse_expression == rse_expression)
-        )
-        global_account_limit = session.execute(stmt).scalar_one()
-        if global_account_limit.bytes == -1:
-            return float("inf")
-        else:
-            return global_account_limit.bytes
-    except NoResultFound:
-        return None
-
-
-@read_session
-def get_local_account_limit(account: "InternalAccount", rse_id: str, *, session: "Session") -> Union[int, float, None]:
-    """
-    Returns the account limit for the account on the rse.
-
-    :param account:  Account to check the limit for.
-    :param rse_id:   RSE id to check the limit for.
-    :param session:  Database session in use.
-    :return:         Limit in Bytes.
-    """
-    try:
-        stmt = select(
-            models.AccountLimit
-        ).where(
-            and_(models.AccountLimit.account == account,
-                 models.AccountLimit.rse_id == rse_id)
-        )
-        account_limit = session.execute(stmt).scalar_one()
-        if account_limit.bytes == -1:
-            return float("inf")
-        else:
-            return account_limit.bytes
-    except NoResultFound:
-        return None
-
-
-@read_session
-def get_local_account_limits(account: "InternalAccount", rse_ids: Optional[list[str]] = None, *, session: "Session") -> dict[str, int]:
+def get_local_account_limit(account: "InternalAccount", rse_ids: Union[str, list[str], None] = None, *, session: "Session") -> Union[int, float, dict[str, int], None]:
```

Comment:
```suggestion
def get_local_account_limit(account: "InternalAccount", rse_ids: Optional[str, 'Iterable[str]'] = None, *, session: "Session") -> Optional[int, float, dict[str, int]]:
```

1. Prefer `Optional` instead of `Union[..., None]`
2. `rse_ids` is used as an `Iterable` here, so it does not need to be a list - please update the docstring accordingly as well. In general, when type hinting parameters, it's best to be as generic as possible (i.e. `Iterable` instead of `list`), while it's best to be more specific when type hinting return types

Reply:
Thanks! I’ll keep that in mind.  

Just out of curiosity, is using `Optional` instead of `Union[..., None]` just a convention in the Rucio codebase, or does it have any benefits in terms of performance or readability? For me, they're both same and `Union[..., None]` offers backward compatibility in most old codebases. We might also switch to https://docs.python.org/3/library/typing.html#typing.Optional `str | None`.

Reply:
It's a convention in the Rucio codebase - it doesn't change anything performance-wise, and I guess it's subjective whether it's more or less readable. 

> Union[..., None] offers backward compatibility in most old codebases.

I mean, that's not really something we would need to worry about, we're on Python 3.9+.

> We might also switch to https://docs.python.org/3/library/typing.html#typing.Optional str | None.

That was added in Python 3.10, and we need to maintain compatibility with Python 3.9 for now.